tests/test_input_output.py

```diff
@@ -111,6 +111,103 @@ def test_write_desc_input_Nones(tmpdir_factory):
     assert no_gtol  # fail test if gtol line is written
 
 
+@pytest.mark.unit
+def test_descout_to_input(tmpdir_factory):
+    """
+    Test converting DESC output to a DESC input file.
+
+    To do that, we convert a DESC output file to a DESC input file
+    named desc_input_converted. We compare the boundary Fourier
+    coefficients of this file with the true values of the Fourier
+    coefficients given in desc_input_truth.
+    """
+    outfile_path = "./tests/inputs/LandremanPaul2022_QA_reactorScale_lowRes.h5"
+    tmpdir = tmpdir_factory.mktemp("desc_inputs")
+    tmp_path = tmpdir.join("input_LandremanPaul_QA")
+    tmpout_path = tmpdir.join("LandremanPaul2022_QA_reactorScale_lowRes.h5")
+    shutil.copyfile(outfile_path, tmpout_path)
+
+    ir1 = InputReader()
+    ir1.descout_to_input(str(tmp_path), str(tmpout_path))
+    ir1 = InputReader(cl_args=[str(tmp_path)])
+    arr1 = ir1.parse_inputs()[-1]["surface"]
+    arr1 = arr1[arr1[:, 1].argsort()]
+    arr1mneg = arr1[arr1[:, 1] < 0]
+    arr1mpos = arr1[arr1[:, 1] >= 0]
+    pres1 = ir1.parse_inputs()[-1]["pressure"]
+
+    desc_input_truth = "./tests/inputs/LandremanPaul2022_QA_reactorScale_lowRes"
+    with pytest.warns(UserWarning):
+        ir2 = InputReader(cl_args=[str(desc_input_truth)])
+        arr2 = ir2.parse_inputs()[-1]["surface"]
+        pres2 = ir2.parse_inputs()[-1]["pressure"]
+    arr2 = arr2[arr2[:, 1].argsort()]
+    arr2mneg = arr2[arr2[:, 1] < 0]
+    arr2mpos = arr2[arr2[:, 1] >= 0]
+
+    np.testing.assert_allclose(
+        np.minimum(
+            np.linalg.norm(arr1mneg[:, 3:] - arr2mneg[:, 3:]),
+            np.linalg.norm(arr1mneg[:, 3:] + arr2mneg[:, 3:]),
+        ),
+        0,
+        atol=1e-8,
+    )
+    np.testing.assert_allclose(
+        np.minimum(
+            np.linalg.norm(arr1mpos[:, 3:] - arr2mpos[:, 3:]),
+            np.linalg.norm(arr1mpos[:, 3:] + arr2mpos[:, 3:]),
+        ),
+        0,
+        atol=1e-8,
+    )
+
+    if np.linalg.norm(pres1[:, 1]) > 0:
```

Comment:
is there any reason for the conditional here?

Reply:
Only match pressure profiles with nonzero coefficients.
No need to match vacuum pressures.